collect/cnrfc/cnrfc.py

```diff
@@ -318,6 +318,65 @@ def get_deterministic_forecast_watershed(watershed, date_string, acre_feet=False
                                  'downloaded': dt.datetime.now().strftime('%Y-%m-%d %H:%M')}}
 
 
+def get_deterministic_forecast_watershed_IO(watershed, date_string, path):#, cnrfc_id=None):
+    """
+    from: https://www.cnrfc.noaa.gov/deterministicHourlyProductCSV.php
+    https://www.cnrfc.noaa.gov/csv/2019040318_american_csv_export.zip
+
+    Arguments:
+        watershed (str):
+        date_string (str):
+        path (str): file name including file path/s3 key
+        cnrfc_id (str): 
+    Returns:
+        (dict): 
+    """
+    units = 'kcfs'
+
+    # store original date_string
+    _date_string = date_string
+
+    # forecast datestamp prefix
+    date_string = _default_date_string(date_string)
+
+    # data source
+    url = 'https://www.cnrfc.noaa.gov/csv/{0}_{1}_csv_export.zip'.format(date_string, watershed)
+
+    # extract CSV from zip object
+    if get_web_status(url):
+        try:
+            csvdata = _get_forecast_csv(url)
+        except zipfile.BadZipFile:
+            print(f'ERROR: forecast for {date_string} has not yet been issued.')
+            raise zipfile.BadZipFile
+    
+    # raise error if user supplied an actual date string but that forecast doesn't exist
+    elif _date_string is not None:
+        print(f'ERROR: forecast for {date_string} has not yet been issued.')
+        raise zipfile.BadZipFile
+
+    # forecast issue time
+    time_issued = get_watershed_forecast_issue_time('hourly', watershed, date_string, deterministic=True)
+
+    # set path for 
+    if path is None:
+        datestr = url.split('_')[0].split('/')[-1]
+        watershedstr = url.split('_')[1]
+        path = '{}/{}_{}_csv_export.csv'.format(os.getcwd(), datestr, watershedstr)
+
+    # write csvdata to specified path
+    path = path.replace('/', os.sep)
```

Comment:
Let's simplify definition of the default path by linking it to the exact pattern in the url.  Could use something like this:
```python
if path is None:
    path = url.split('/')[-1].replace('.zip', '.csv')
```
This would work for both deterministic and ensemble watershed forecasts.

Reply:
I incorporated the suggestion and set the path default to None. 

Resolved in 575ffcdf91cd4666640f7435348bce6ad6e611b6.